File: apps/cargo-tracker/src/main/java/com/example/cargotracker/routing/domain/model/CargoType.java
package com.example.cargotracker.routing.domain.model;

/**
 * routing コンテキストにおける貨物種別。
 */
public enum CargoType {
    /** 一般貨物 */
    GENERAL("一般貨物"),
    /** 危険物 */
    HAZARDOUS("危険物"),
    /** 冷凍貨物 */
    REFRIGERATED("冷凍・冷蔵");

    private final String displayName;

    CargoType(String displayName) {
        this.displayName = displayName;
    }

    public String getDisplayName() {
        return displayName;
    }
}

File: docs/article/source/scala/apps/cargo-tracker/app/cargotracker/shared/domain/CargoType.scala
package cargotracker.shared.domain

/** 貨物種別。料金計算の係数決定に利用する。 */
enum CargoType:
  case General
  case Hazardous
  case Refrigerated

object CargoType:
  def fromName(name: String): Option[CargoType] =
    values.find(_.toString.equalsIgnoreCase(name))
